src/icastats_shared.c

```diff
@@ -34,18 +34,18 @@ static stats_entry_t *stats = NULL;
 volatile int stats_shm_handle = NOT_INITIALIZED;
 
 
-static inline void atomic_add(int *x, int i)
+static inline void atomic_add(uint64_t *x, uint64_t i)
 {
-	int old;
-	int new;
-	asm volatile ("	l	%0,%2\n"
-		      "0:	lr	%1,%0\n"
-		      "	ar	%1,%3\n"
-		      "	cs	%0,%1,%2\n"
-		      "	jl	0b"
-		      :"=&d" (old), "=&d"(new), "=Q"(*x)
-		      :"d"(i), "Q"(*x)
-		      :"cc", "memory");
+    uint64_t old;
+    uint64_t new;
+    asm volatile ("   lg      %0,%2\n"
+                  "0: lgr     %1,%0\n"
+                  "   agr     %1,%3\n"
+                  "   csg     %0,%1,%2\n"
```

Comment:
isn't this CSD, not CDG ?

Reply:
well, CSD swaps 8 bytes, but uses 32 bit registers. So maybe CSG is the right one....

Reply:
Yes, CSG should be the right one. Maybe Patrick can finally confirm. 
BTW: I also replaced tabs by spaces, so that the operands start after 8 chars including the instruction mnemonic. In libica a tab is only four spaces. However, it didn't cause a problem before.

Reply:
imo you need csg here.